src/dataEditor/dataEditorClient.ts

```diff
@@ -119,7 +120,11 @@ export function activate(ctx: vscode.ExtensionContext): void {
       DATA_EDITOR_COMMAND,
       async (fileToEdit: string = '') => {
         let configVars = editor_config.extractConfigurationVariables()
-        return await createDataEditorWebviewPanel(ctx, configVars, fileToEdit)
+        return await createDataEditorWebviewPanel(
+          ctx,
+          configVars,
+          vscode.debug.activeDebugSession ? getCurrentConfig().data : fileToEdit
```

Comment:
There is a function for checking the debug session status in _src/dataEditor/include/utils.ts_
https://github.com/apache/daffodil-vscode/blob/a6f1cbcb7c7fbdb50f05aa8b3ac141192c056ce6/src/dataEditor/include/utils.ts#L19-L24